### PR TITLE
Directory initial creation and json api

### PR DESCRIPTION
**Pull request by imsickofmaps**
## Basics
- Category name (up to 60 characters)
- Link to sub-category, one to many
- Sub-categories have a name (up to 60 characters) and 1-3 parts description
- The three parts are up to 125 characters in length
## API

There needs to be only one endpoint

**GET /api/directory/**

```
{
"directory": {
    "category one": {
        "sub category one": {
            "description_1": "first part of contact details",
            "description_2": "second part of contact details",
            "description_3": "third part of contact details"          
        },
        "sub category two": {
            "description_1": "first part of contact details",
            "description_2": "second part of contact details",
            "description_3": "third part of contact details"          
        }
    },
    "category two": {
        "sub category one": {
            "description_1": "first part of contact details",
            "description_2": "second part of contact details",
            "description_3": "third part of contact details"          
        },
        "sub category two": {
            "description_1": "first part of contact details",
            "description_2": "second part of contact details with none extra"        
        }
    }
}
}
```

### CODE REVIEW

File: directory/admin.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    def clean(self):
+        super(SubCategoryFormset, self).clean()
+
+        for form in self.forms:
+            if not hasattr(form, 'cleaned_data'):
+                continue
+
+            sub_cat_index = 0
+            if ("name" not in form.cleaned_data and sub_cat_index >= 1):
+                raise forms.ValidationError("You need the Sub Category Name")
```
Comment (imsickofmaps) [inline directory/admin.py]: @gitumarkk please change to "You need to name the Sub Category"

File: directory/admin.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+            if ("name" in form.cleaned_data):
+                if not (form.cleaned_data["content_1"] or
+                        form.cleaned_data["content_2"] or
+                        form.cleaned_data["content_2"]):
+                    raise forms.ValidationError("You need atleast 1 Content")
```
Comment (imsickofmaps) [inline directory/admin.py]: @gitumarkk please change to "You need to enter at least one screen of content to display"

File: directory/api.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        allowed_methods = ['get']
+        excludes = ['completed', 'active']
+        include_resource_uri = False
+        queryset = Category.objects.all()
+
+    def alter_list_data_to_serialize(self, request, data_dict):
+        """
+        Modifying the data to provide the quiz and answers and responses.
+        Structure is {dir: {cat: {sub_cat{c: "a", c2: "b", "c3: "c"}}}}
+        This function handles /api/weeklyquiz/
```
Comment (imsickofmaps) [inline directory/api.py]: @gitumarkk ?

File: directory/tests.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+"""
+This file demonstrates writing tests using the unittest module. These will pass
+when you run "manage.py test".
+
+Replace this with more appropriate tests for your application.
+"""
+
+from django.test import TestCase
```
Comment (imsickofmaps) [inline directory/tests.py]: @gitumarkk remove stub

**Comment by imsickofmaps** [comment]
@gitumarkk just a few tweaks then @ me and I'll re-review and push over to PRK

File: directory/api.py (hunk 3 above)
Comment (gitumarkk) [inline directory/api.py]: Not sure what you mean?

Comment (imsickofmaps) [inline directory/api.py]: This is not the weekly quiz! Its the directory...

**Comment by gitumarkk** [comment]
@imsickofmaps I have made the required changes for all the issues

**Comment by imsickofmaps** [comment]
@smn ready for your review

File: directory/admin.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+                continue
+
+            sub_cat_index = 0
+            if ("name" not in form.cleaned_data and sub_cat_index >= 1):
+                raise forms.ValidationError("You need to name the Sub Category")
+
+            if ("name" in form.cleaned_data):
+                if not (form.cleaned_data["content_1"] or
+                        form.cleaned_data["content_2"] or
+                        form.cleaned_data["content_2"]):
```
Comment (smn) [inline directory/admin.py]: checking for `content_2` twice

**Pull request merged**